### PR TITLE
feat(property): Allow user/data curator to specify the avatar initials for organizations (backend)

### LINKED ISSUE #1481: [Story] Allow user/data curator to specify the avatar initials for organizations
### What projects is this story for?

OpenChallenges

### As a user, I want

As a user/data curator, I want to be able to customize the initials displayed in the avatar (when no image is available) so that I can prevent "mistakes" made when inferring the initials from the name of the organization.

The same could apply to users when we resume working on the user implementation.

### Description

_No response_

### Acceptance criteria

Out of scope:

- Frontend code would be updated in a separate PR, including updating the API client for Angular.

### Tasks

- [x] Add the property `organization.acronym` to the API description
- [ ] Add the property `organization.acronym` to the data model
- [x] Update the organization service to support the new parameter
- [ ] #1520

### Anything else?

- This is a good task for Maria as part of her onboarding.
  - Minimal change made to the API description + update of one microservice
  - Collaboration with another developer in the same PR (Verena will update the data)
- This feature was suggested by Solly during the April sprint review meeting.

### Have you linked this story to a GitHub Project?

- [X] I have linked this story to a GitHub Project and set its metadata.

### PR DESCRIPTION
**Pull request by mdsage1**
Contributes to #1481

## Changelog

- update the API description of the organization service to add the property Organization.acronym
- update the data model of the organization service (SQL)

## Notes

- The acronym feature has been added as a property of organization service. Future development may include acronym optimization using the organization name being compared to the available acronym.

### The organization service now returns `Organization.acronym`

The motivation for this change is that the challenge search page will use the organization acronym in the `ChallengeSearchQuery` object to filter challenges by organization. This will serve as a secondary organization identifier and will eliminate the use of a shortName descriptor.

#### Addition of acronym to all SQL queries.

```
{
  "name": "University of Washington",
  "email": "",
  "login": "uw",
  "description": "",
  "avatarUrl": "http://localhost:8082/img/PDzla5KBEuTCo4bTd6_X3yznCk4=/300x300/logo/University_of_Washington.svg",
  "websiteUrl": "https://www.washington.edu/",
  "challengeCount": 7,
  "createdAt": "2023-05-08T21:24:21Z",
  "updatedAt": "2023-05-08T21:24:21Z",
  "acronym": "UW"
}
```

### CODE REVIEW

File: apps/openchallenges/organization-service/src/main/resources/db/migration/V1.0.1__insert_temp_data.sql [hunk 1]
```diff
@@ -8,7 +8,8 @@ INSERT INTO organization (
     avatar_url,
     website_url,
     description,
-    challenge_count
+    challenge_count.
```
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/main/resources/db/migration/V1.0.1__insert_temp_data.sql:11]:
```suggestion
    challenge_count,
```

File: libs/openchallenges/api-description/src/components/schemas/Organization.yaml [hunk 2]
```diff
@@ -42,3 +45,4 @@ required:
   - websiteUrl
   - createdAt
   - updatedAt
+  - acronym
```
Comment (tschaffter) [inline libs/openchallenges/api-description/src/components/schemas/Organization.yaml:48]: I would make this property optional as not all orgs may have an acronym.

File: apps/openchallenges/organization-service/src/main/resources/db/migration/V1.0.0__create_tables.sql [hunk 3]
```diff
@@ -11,7 +11,6 @@ CREATE TABLE `organization` (
   `challenge_count` int,
   `created_at` DATETIME DEFAULT CURRENT_TIMESTAMP,
   `updated_at` DATETIME DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
-  PRIMARY KEY (`id`),
```
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/main/resources/db/migration/V1.0.0__create_tables.sql:14]: Does this line need to be removed?

File: apps/openchallenges/organization-service/src/main/resources/db/migration/V1.0.0__create_tables.sql
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/main/resources/db/migration/V1.0.0__create_tables.sql:1]: The new property should be added to the table `organization`

**Review by tschaffter — CHANGES_REQUESTED** [review]
This is a great start. See the comments I left.

There is one more file that should be updated in the org service, see `model/entity/OrganizationEntity.java`.

**Comment by tschaffter** [comment]
@mdsage1 In the Original Post (OP), could you add a section named "Preview" that shows an example of Organization object returned by the organization service to illustrate this contribution?

**Comment by mdsage1** [comment]
The VALUES have not been updated in the INSERT INTO organization query. They will be updated soon.

**Comment by tschaffter** [comment]
@mdsage1 The current snippet of code in the section `Preview` is not required because reviewers can see these changes under the tab `Files changed`. What I suggested was to show the JSON response returned by the organization service. This proves that the implementation works. You can get this response from the Swagger UI or from the file `requests.http` that lives in the project folder of the microservices.

**Comment by mdsage1** [comment]
The requested change has been implemented and pushed to this repo

**Pull request merged**